### PR TITLE
refactor: replace non output EventEmitter with Subject. 

### PR DESCRIPTION
**Pull request by JeanMeche**
2 commits, 1 form the common package, 1 for the core package. 

Same as #54665

### CODE REVIEW

File: packages/common/src/location/location.ts [hunk 1]
```diff
@@ -265,7 +265,7 @@ export class Location implements OnDestroy {
     onThrow?: ((exception: any) => void) | null,
     onReturn?: (() => void) | null,
   ): SubscriptionLike {
-    return this._subject.subscribe({next: onNext, error: onThrow, complete: onReturn});
+    return this._subject.subscribe({next: onNext, error: onThrow!, complete: onReturn!});
```
Comment (devversion) [inline packages/common/src/location/location.ts:268]: Do you want to use `onThrow ?? undefineed`?

**Review by devversion — APPROVED** [review]
🎉  related to #54650

File: packages/common/src/location/location.ts (hunk 1 above)
Comment (JeanMeche) [reply packages/common/src/location/location.ts]: Yes, it's more reasonable. Thank you for the suggestion.

**Review by pkozlowski-opensource — APPROVED** [review]

**Review by atscott — APPROVED** [review]

**Comment by pkozlowski-opensource** [comment]
This PR was merged into the repository by commit 5a04837c54bcf2044676feafe126b00dd3b19a5c.

The changes were merged into the following branches: main

**Pull request closed without merging**